manager/config/config.go

```diff
@@ -0,0 +1,62 @@
+package config
+
+import (
+	"time"
+
+	"github.com/flyteorg/flytestdlib/config"
+)
+
+//go:generate pflags Config --default-var=DefaultConfig
+//go:generate enumer --type=ShardType --trimprefix=ShardType -json -yaml
+
+var (
+	DefaultConfig = &Config{
+		PodApplication:           "flytepropeller",
+		PodTemplateContainerName: "flytepropeller",
+		PodTemplateName:          "flytepropeller-template",
+		PodTemplateNamespace:     "flyte",
+		ScanInterval: config.Duration{
+			Duration: 10 * time.Second,
+		},
+		ShardConfig: ShardConfig{
+			Type:       ShardTypeHash,
+			ShardCount: 3,
+		},
+	}
+
+	configSection = config.MustRegisterSection("manager", DefaultConfig)
+)
+
+type ShardType int
+
+const (
+	ShardTypeDomain ShardType = iota
+	ShardTypeProject
+	ShardTypeHash
+)
+
+// Configuration for defining shard replicas when using project or domain shard types
+type PerShardMappingsConfig struct {
+	IDs []string `json:"ids" pflag:",The list of ids to be managed"`
+}
+
+// Configuration for the FlytePropeller sharding strategy
+type ShardConfig struct {
+	Type             ShardType                `json:"type" pflag:"-,Shard implementation to use"`
```

Comment:
If you have the latest pflags binary, you should be able to generate binding for this... does it error?

```suggestion
	Type             ShardType                `json:"type" pflag:",Shard implementation to use"`
```

Reply:
done